src/sources/journald.rs

```diff
@@ -1014,7 +1025,25 @@ mod tests {
 
     #[tokio::test]
     async fn handles_acknowledgements() {
-        let (tx, mut rx) = SourceSender::new_test();
+        let (count, checkpoint) = run_acknowledgements(usize::MAX).await;
+
+        assert_eq!(count, 8);
+        assert_eq!(checkpoint.as_deref(), Some("8"));
+    }
+
+    #[tokio::test]
+    async fn handles_negative_acknowledgements() {
+        let (_count, checkpoint) = run_acknowledgements(2).await;
+
+        assert_eq!(checkpoint.as_deref(), Some("2"));
+        // The acknowledgements for the events are delivered after the
+        // events are delivered to the pipeline, so this test would
+        // fail to show that deliveries have stopped.
+        // assert_eq!(count, 2);
+    }
```

Comment:
One thing it _feels_ like this test is missing is positively acknowledging an event _after_ the initial negative acknowledgement, to assert that we correctly stop processing no matter what.